langchain4j-mcp/src/main/java/dev/langchain4j/mcp/client/McpClient.java

```
@@ -47,4 +47,10 @@ public interface McpClient extends AutoCloseable {
      * Render the contents of a prompt.
      */
     McpGetPromptResult getPrompt(String name, Map<String, Object> arguments);
+
+    /**
+     * Performs a health check that returns true if the MCP server is reachable and
```

Javadoc does not correspond to the implementation.

How do you think this method will be used? Perhaps it should indeed return `true/false` instead of throwing exception? Exception-driven logic does not feel right.

---

We also need to pass an error message to say what went wrong. Just `true/false` isn't enough information. 

Ideally we could return the full `HealthCheckResponse` (https://github.com/microprofile/microprofile-health/blob/main/api/src/main/java/org/eclipse/microprofile/health/HealthCheckResponse.java) that carries all the proper information, but that would mean introducing a MP Health dependency - not sure if you want this.

So the approach I thought of was that we let it throw an exception and then the wrapping framework (Quarkus, or something that the user writes himself) would transform that into a `HealthCheckResponse` (or something else if using something other than MP Health).

But if you don't mind being bound to a specific Health framework, I can redo this by adding MP Health and letting this method return a `HealthCheckResponse`.

---

A full integration with MP Health would require setting up an HTTP endpoint though, so I don't think we should do it. We could, however, introduce our own slimmed down version of `HealthCheckResponse` that would get transformed to the regular one by Quarkus?! That way we wouldn't depend on MP Health here.

---

Both "our own slimmed down version of HealthCheckResponse" and exception seem fine, whatever you prefer